ddl/ddl_worker.go

```diff
@@ -534,6 +539,28 @@ func (w *worker) handleDDLJobQueue(d *ddlCtx) error {
 			d.mu.hook.OnJobRunBefore(job)
 			d.mu.RUnlock()
 
+			// allow drop table when disk full
+			switch job.Type {
+			case model.ActionDropSchema:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropTable:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropIndex:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionTruncateTable:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropTablePartition:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropView:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropSequence:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
+			case model.ActionDropIndexes:
+				txn.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlreadyFull)
```

Comment:
Same branch for several cases. Is it possible to merge them into one?

Reply:
Done